### PR TITLE
[SPARK-19446][SQL] Remove unused findTightestCommonType in TypeCoercion

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes to 

- remove unused `findTightestCommonType` in `TypeCoercion` as suggested in https://github.com/apache/spark/pull/16777#discussion_r99283834
- rename `findTightestCommonTypeOfTwo ` to `findTightestCommonType`.
- fix comments accordingly

The usage was removed while refactoring/fixing in several JIRAs such as SPARK-16714, SPARK-16735 and SPARK-16646

## How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @gatorsmile

**Comment by gatorsmile** [comment]
Could you rename `findTightestCommonTypeOfTwo` to `findTightestCommonType`?

**Comment by gatorsmile** [comment]
Also clean up the comments?

**Comment by HyukjinKwon** [comment]
Thanks, I just updated and rebased.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala [hunk 1]
```diff
@@ -253,7 +253,7 @@ private[sql] object InferSchema {
    * Returns the most general data type for two given data types.
    */
   def compatibleType(t1: DataType, t2: DataType): DataType = {
-    TypeCoercion.findTightestCommonTypeOfTwo(t1, t2).getOrElse {
+    TypeCoercion.findTightestCommonType(t1, t2).getOrElse {
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala]: In CSVInferSchema.scala, it sounds like we are also calling the same function. Could you check whether we can directly call it, instead of duplicating the code? (I did not read it carefully)

https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchema.scala#L172-L217

Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala]: Oh, yea, it appearntly possible for now. But, can I maybe do this in another PR if you think it is fine? It seems possible but I would like to avoid to fix it here simply because of a worry to resolve a conflict in the PR refacorting this code path and it seems not directly related to the JIRA.

Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala]: Sure, no problem. Please submit a separate PR for it. Thanks!

Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala]: Thank you for bearing with me.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by gatorsmile** [comment]
LGTM pending test

**Comment by gatorsmile** [comment]
Thanks, merging to master!

**Comment by HyukjinKwon** [comment]
Thank you @gatorsmile.
